mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogEventPool.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Log {
+
+	sealed class LogEventPool {
```

Comment:
@marek-safar The problem is a lot more complicated.

The log *must* be sequentially read, performance tanks otherwise, even on SSDs. Random seeking kill all forms of buffering.

Events must be buffered until we reach a sync point and at that point we sort them. Using value types for the decoded payload would be horribly inefficient as events have an unbound max size or we'd be forced to decode them twice - and the second time would require seeking.

I don't know what you mean by sort at insertion time, there's no inserting time. If you mean at decoding time. That's even more inefficient that sorting at sync points. Remember, we don't report events until we reach a sync point.

Finally, APIs have being using FlyWeight objects since the dawn of time, as long as it's well documented, I think it's fine.